kajas/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,123 @@
+          <input type="email" id="email" formControlName="email" 
+            placeholder="Email Address *" required>
+              @if(registrationForm.get('email').invalid && 
+                registrationForm.get('email').touched) {
+                  <div class="error-message">
+                    {{ getErrorMessage('email') }}
+                  </div>
+              }
+        </div>
+      </div>
+      <div class="form-group-wrapper password-container">
+        <div class="form-group">
+          <input [type]="hidePassword ? 'password' : 'text'" id="password" 
+            formControlName="password" placeholder="Password *" required>
+          <img src="./assets/icons/{{hidePassword ? 'unsee' : 'see'}}.png" 
+            alt="Toggle Password Visibility" class="toggle-password"
+            (click)="togglePasswordVisibility('password')">
+              @if(registrationForm.get('password').invalid && 
+                registrationForm.get('password').touched) {
+                  <div class="error-message">
+                    {{ getErrorMessage('password') }}
+                  </div>
+              }
+        </div>
+      </div>
+      <div class="form-group-wrapper password-container">
+        <div class="form-group">
+          <input [type]="hideConfirmPassword ? 'password' : 'text'" 
+            id="confirm_password" formControlName="confirmPassword" 
+              placeholder="Confirm Password *" required>
+          <img src="./assets/icons/{{hideConfirmPassword ? 'unsee' : 'see'}}.png"
+            alt="Toggle Password Visibility" class="toggle-password" 
+            (click)="togglePasswordVisibility('confirmPassword')">
+              @if(registrationForm.get('confirmPassword').invalid &&
+                registrationForm.get('confirmPassword').touched) {
+                  <div class="error-message">
+                    {{ getErrorMessage('confirmPassword') }}
+                  </div>
+              }
+        </div>
+      </div>
+      <div class="form-group-wrapper">
+        <div class="form-group">
+          <input type="text" id="first_name" 
+            formControlName="firstName" placeholder="First Name *" required>
+              @if(registrationForm.get('firstName').invalid && 
+                registrationForm.get('firstName').touched) {
+                  <div class="error-message">
+                    {{ getErrorMessage('firstName') }}
+                  </div>
+                }
+        </div>
+      </div>
+      <div class="form-group-wrapper">
+        <div class="form-group">
+          <input type="text" id="middle_name" formControlName="middleName" 
+            placeholder="Middle Name">
+        </div>
+      </div>
+      <div class="form-group-wrapper">
+        <div class="form-group">
+          <input type="text" id="last_name" formControlName="lastName" 
+            placeholder="Last Name *" required>
+              @if(registrationForm.get('lastName').invalid && 
```

Comment:
### For template control flow, observe proper indentation
- Try to find a [solution](https://github.com/angular/angular/issues/52720) online or uninstall prettier and format your codes manually.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_15